src/_pytest/python.py

```diff
@@ -1356,7 +1356,10 @@ def write_docstring(tw, doc, indent="    "):
             tw.write(indent + line + "\n")
 
 
-class Function(FunctionMixin, nodes.Item, fixtures.FuncargnamesCompatAttr):
+# Ignore type because mypy is confused about Node.repr_failure is defined.
```

Comment:
is this because the mixin causes a conflict here? I'm not sure I understand the comment :S

Reply:
I think the types match but the way `Node.repr_failure` is defined confused mypy:

https://github.com/pytest-dev/pytest/blob/2180d9ef6d9af45c80fdc89cd64b90df2924c8a7/src/_pytest/nodes.py#L238-L283

But if I replace the assignment with a forwarding method it works out, so I'll do that.